enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,179 @@
+  - "@jerolimov"
+  - "@cvogt"
+creation-date: 2022-07-27
+last-updated: 2022-08-23
+tracking-link: 
+  - https://issues.redhat.com/browse/ODC-6695
+status: implementable
+---
+
+# Customize Catalogs
+
+## Summary
+
+The OpenShift Console provides a "Developer Catalog" which enables 
+the user to create apps and services based on Operator backed services, helm charts, templates, etc.
+
+Cluster admins can provide additional catalog items by installing operators or adding helm repositories.
+Cluster administrators want to hide these types.
+
+Under the developer catalog,
+there are categories and sub-categories. When users select a sub-category, they can see the list
+of types to install. This enhancement enable cluster admins to customize these types.
+
+![Developer catalog and types list](customize-catalogs.png)
+ 
+## Motivation
+
+Cluster administrators don't want the users to use the catalog 
+to install anything in the cluster until
+permission is given. Cluster administrators want to 
+disable developer catalog and types for all users.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide Developer catalog for all users.
+
+#### Story 2
+
+As a cluster admin, I want to hide types for all users.
+
+### Goals
+
+- Allow cluster admins to hide developer catalog and types.
+- Provide information to admin about types they can hide from users.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`. 
+We can add 
+`spec.customization.developerCatalog.disableTypes` which can have list of types disabled.
+
+### Workflow Description
+
+```text
+├── spec
+│   ├── customization
+|       ├── developerCatalog 
+|           ├── disableTypes
+|               └──       
+└── ...
+```
+
+Each catalog conforms to the following schema:
+
+// +optional
+- `disableTypes: string[]` , defines a list of types to hide. 
+Types will vary based on the operator installed. Cluster administrator will be provided with 
+the snippet about the available types.
+
+### API Extensions
+
+Extend the existing `Console` CRD to add `spec.customization.developerCatalog.disableTypes` which 
+can have list of types disabled. 
+
+So the example yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec: 
+  customization:
+    developerCatalog:
+      disableTypes: 
+        - Devfile
+        - HelmChart
+  ...
+```
+
+If all the available types are listed in `spec.customization.developerCatalog.disableTypes` 
+then complete developer catalog will be disabled. A snippet will be provided to the cluster administrator
+that provide details about the available types.
+
+### Risks and Mitigations
+
+**Save selected**: When we save selected types, new type added 
+by an operator will not be shown until the cluster administrator 
+updates the types list.
+This is similar to Developer Catalog categories and sub-categories! 
```

Comment:
Is this still true? We are configuring only the disabled types, correct?

Reply:
Yes.. removed it.